src/main/java/org/jenkinsci/plugins/workflow/support/actions/WorkspaceRunAction.java

```diff
@@ -59,19 +59,20 @@ public final class WorkspaceRunAction implements Action {
         this.owner = owner;
     }
 
+    private boolean canAccess() {
+        return (build instanceof AccessControlled && !((AccessControlled) build).hasPermission(Item.WORKSPACE));
```

Comment:
This might be a nit, but when I think of `canAccess`, I would think that a `true` would mean that I do have access. In this case, `canAccess` is returning `false`, it feels a bit confusing for me.

Reply:
I've renamed the method.

Reply:
Sorry, I think I am confusing myself. Please correct me if I am wrong here:

In the code here, the method is returning `true` when there is no workspace permission. But wouldn't the name `hasWorkspacePermission` or `canAccess` mean that when it returns `true` the user has permission?

Unless I am mistaken and that you actually do return `null` when the user has workspace permission?

Reply:
It's the same condition used before to check whether the user has access to the workspace, just wrapped into a private method to reuse a bunch of times.

The code does return false if the user lacks permission to access the workspace or build is no instance of AccessControlled, and in this case the icon to access the workspace is hidden from the UI.